### PR TITLE
Added example for limiting workflow concurrency in .NET fan-in/out example

### PR DESCRIPTION
**Pull request by WhitWaldo**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
I love rich documentation. It's so satisfying to read something that details a new concept, shares examples and really sets me up to succeed. But I don't like it when said documentation introduces another idea and leaves me to my own devices to figure out how to do it.

Tonight I was reading about the different workflow patterns. In the fan-out/fan-in pattern, it explains how a series of parallel tasks can be completed at once, but then it leaves a note at the end: "While not shown in the example, it's possible to go further and limit the degree of concurrency using simple, language-specific constructs."

As it took me a minute to figure out a worthwhile .NET approach to doing this, I wanted to contribute it back so future me isn't left hanging. I leave it to others to figure out other language equivalents.

## Issue reference

None - saw this in the docs and after determining a solution, wanted to contribute it back.

### CODE REVIEW

**Comment by hhunter-ms** [comment]
Thanks for this contribution @WhitWaldo!

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md [hunk 1]
```diff
@@ -287,7 +287,65 @@ The key takeaways from this example are:
...
+{
+  public static async Task<IEnumerable<T>> WhenAllWithLimitAsync<T>(this IEnumerable<T>> tasks, int maxDegreeOfParallelism)
+  {
+    var semaphore = new SemaphoreSlim(maxDegreeOfParallelism);
+    var tasksWithSemaphore = new List<Task<T>>();
+    
+    foreach(var task in tasks)
+    {
+      //Wait for the semaphore to become available
+      await semaphore.WaitAsync();
```
Comment (cgillum) [inline daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md]: This actually violates one of the workflow code constraints, which is that you mustn't await any task that isn't created by the context object, so unfortunately we can't use `SemaphoreSlim` for this. Instead, the right way to do this is to follow an approach similar to what is posted here for Durable Functions: https://stackoverflow.com/a/58794510/2069.

**Review by cgillum — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md (hunk 1 above)
Comment (WhitWaldo) [reply daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md]: Haha, whoops - further illustrating why it's frustrating for the docs to leave this as an exercise to the reader to do properly. I'll review your answer there and adapt it for a second attempt. Thank you!

Comment (cgillum) [reply daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md]: Haha, that's a great point. 😅 Looking forward to the update! This is one of those questions we get a lot and I always kick myself for not having it properly documented somewhere. Thanks for doing this.

**Comment by hhunter-ms** [comment]
@WhitWaldo friendly bump

**Comment by WhitWaldo** [comment]
@hhunter-ms Forgot all about this - I'll see if I can't get something working soon. Thanks!

**Comment by WhitWaldo** [comment]
@hhunter-ms Updated the docs to reflect the example given in the SO post to include support for retaining a returned value for each of the tasks

**Comment by hhunter-ms** [comment]
@WhitWaldo it looks like 1.13 was pulled into this PR, which is pointing into 1.12 - do you mind either reopening as a new PR, or somehow reverting the 1.13 merge?

**Pull request closed without merging**